ui/lib/core/addon/components/form-field-groups.js

```diff
@@ -22,24 +22,21 @@ import layout from '../templates/components/form-field-groups';
  *  @validationMessages={{validationMessages}}
  * />
  * ```
- *
- * @param [renderGroup=null] {String} - An allow list of groups to include in the render.
- * @param model=null {DS.Model} - Model to be passed down to form-field component. If `fieldGroups` is present on the model then it will be iterated over and groups of `FormField` components will be rendered.
- * @param onChange=null {Func} - Handler that will get set on the `FormField` component.
- * @param onKeyUp=null {Func} - Handler that will set the value and trigger validation on input changes
- * @param validationMessages=null {Object} Object containing validation message for each property
+ * @callback onChangeCallback
+ * @callback onKeyUpCallback
+ * @param {Model} model- Model to be passed down to form-field component. If `fieldGroups` is present on the model then it will be iterated over and groups of `FormField` components will be rendered.
+ * @param {string} [renderGroup] - An allow list of groups to include in the render.
+ * @param {onChangeCallback} [onChange] - Handler that will get set on the `FormField` component.
```

Comment:
these look like duplicates of the above on row 25 and 26?

Reply:
The pattern is to define the callback using `@callback` and then use it as a param type. I found this when reviewing the [jsdoc documentation for params](https://jsdoc.app/tags-param.html) under the _Callback Functions_ section at the bottom.

Reply:
Interesting. Nice find.